MFVNeutralino/test/predict2v_newsimpl.py

```diff
@@ -0,0 +1,175 @@
+   alpha_nn[ntk] = alpha
+   sum_rest += (n2v/alpha)
+   sum2_erest += (en2v/alpha)**2 
+   sum_n2v += n2v 
+   sum2_en2v += (en2v**2)
+   sum2_en1v += (en1v**2)
+print 'n1 = %8.0f'%(sum_n1v)
+print 'en1 = %f'%(math.sqrt(sum2_en1v)) 
+
+alpha_nm = {}
+for ntk in 'Ntk3or4','Ntk3or5', 'Ntk4or5':
+   tracks = [int(i) for i in ntk if i.isdigit()]
+   ntktot = sum(tracks)
+   f1 = []
+   for i, n in enumerate(tracks):
+        if n == 5:
+	    tracks[i] = ''
+	    f1.append(fracdict[5]['1v'])
+	else:
+	    tracks[i] = 'Ntk%s' % n
+	    f1.append(fracdict[n]['1v'])
+   n1v0, en1v0 = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % tracks[0]))
+   n1v1, en1v1 = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % tracks[1]))
+   n2v, en2v = get_integral(sel_f.Get('%smfvEventHistosFullSel/h_npu' % ntk))
+   alpha = ((f0 - f1[0])*(f0 - f1[1])) - (f0*(f0-1)) #simplified alphas by assume Cb and Cbbar are about the same for all track multiplicities     
+   alpha_nm[ntk] = alpha
+   sum_rest += (n2v/alpha)
+   sum2_erest += (en2v/alpha)**2
+   sum_n2v += n2v
+   sum2_en2v += (en2v**2) 
+
+for ntk in 3,4,5:
+    n1v, en1v = get_integral(sel_f.Get('%smfvEventHistosOnlyOneVtx/h_npu' % ('' if ntk == 5 else 'Ntk%s' % ntk)))
+    n2v, en2v = get_integral(sel_f.Get('%smfvEventHistosFullSel/h_npu' % ('' if ntk == 5 else 'Ntk%s' % ntk)))
+    n2v_poisson = poisson_interval(n2v)
+    f1 = fracdict[ntk]['1v']
+    cb = cdict[ntk]['cb']
+    cbbar = cdict[ntk]['cbbar']
+    print 'start printing out beta + e_beta'
+    effn1v = n1v/sum_n1v
+    eeffn1v = effn1v * math.sqrt( (en1v/n1v)**2 + ((sum2_en1v)/(sum_n1v**2)))
+    #pred = n1v**2 / (npresel * (1-f0) * (f1 / (1-f1) + 1)**2) * ((f1 / (1 - f1))**2 * ((1 - f0) / f0) * cb + cbbar)
+    pred = (n1v**2 /(sum_n1v)**2) * (alpha_nn[ntk]) * sum_rest  
+    epred = interval_to_vpme(*propagate_ratio(n1v**2, npresel, 2 * n1v * en1v, enpresel))[1] # ignores f0, f1 uncert
+    rat, erat = interval_to_vpme(*propagate_ratio(n2v, pred, en2v, epred))
+    eratl, erath = n2v_poisson / pred
+    #print '%8.0f +- %4.0f %8.3f %8.3f %8.3f %9.3f +- %6.3f %7.1f +- %4.1f  PI: [%5.1f, %5.1f] %7.2f +- %.2f PI: [%4.2f, %4.2f]' % (n1v, en1v, f1, cb, cbbar, pred, epred, n2v, en2v, n2v_poisson[0], n2v_poisson[1], rat, erat, eratl, erath)
+    if print_pred_n2v_propagated_stat_err :
+        # see https://www.evernote.com/shard/s376/nl/201739427/5a38bab1-3b56-442e-82e2-13d9ac597fb8/ ("Stat error on 2-vertex prediction and uncertainty on prediction due to BTV SFs")
+        #pred_n2v_propagated_stat_err = 2 * (n1v * n1v / npresel) * (f1 * f1 / f0) * cb * math.sqrt( 1./(n1v*f1) ) + 2 * (n1v * n1v / npresel) * ( (1-f1) * (1-f1) / (1-f0) ) * cbbar * math.sqrt( 1./(n1v * (1-f1)) )
+        pred_n2v_propagated_stat_err = pred * math.sqrt((eeffn1v/effn1v)**2 + (eeffn1v/effn1v)**2 + (sum2_erest/(sum_rest**2)))
```

Comment:
I haven't gone through the math closely, but is it `sum2_erest/(sum_rest**2) ` or `(sum2_erest/sum_rest)**2`?

Reply:
It is the first because the error was summed up quadratically as in L86.